watertap/examples/flowsheets/RO_with_energy_recovery/parameter_sweep_input_parser.py

```diff
@@ -0,0 +1,310 @@
+                                                values['lower_limit'],
+                                                values['upper_limit'],
+                                                values['num_samples'])
+
+        elif values['type'] == 'UniformSample':
+            sweep_params[param] = UniformSample(child,
+                                                values['lower_limit'],
+                                                values['upper_limit'])
+
+        elif values['type'] == 'NormalSample':
+            sweep_params[param] = NormalSample(child,
+                                               values['mean'],
+                                               values['std'])
+
+        elif values['type'] == 'LatinHypercubeSample':
+            sweep_params[param] = LatinHypercubeSample(child,
+                                                       values['lower_limit'],
+                                                       values['upper_limit'])
+
+    return sweep_params
+
+
+
+def set_defaults_from_yaml(m, yaml_filename, verbose=False, do_nothing=False):
+    """ Sets default model values using values stored in a yaml file
+
+    This function reads a yaml file with the structure::
+
+        m.fs.path.to.attribute_1: 0.123
+        m.fs.path.to.attribute_2: 1.234
+        ...
+
+    and uses the (key, default_value) pairs to set default values
+    for the attributes in model ``m``.
+
+    Args:
+        m (pyomo model):
+            The flowsheet containing the model to set default values for
+        yaml_filename (str):
+            The path to the yaml file.
+
+    Returns:
+        N/A
+
+    """
+
+    input_dict = _yaml_to_dict(yaml_filename)
+
+    fail_ct = 0
+
+    for key, default_value in input_dict.items():
+        # Split the dot-separated object path into a list of
+        # separate objects and attributes
+        path = _split_pyomo_path(key)
+
+        # Follow the attribute path to the correct nested
+        # property, should work with indexing!
+        child = _return_child_property(m, path)
+
+        old_value = value(child)
+
+        if do_nothing:
+            new_value = old_value
+
+        else:
+            new_value = default_value
+
+        if verbose:
+            print('Property: %s' % (key))
+            print('New Value: %e, Old Value: %e' % (new_value, old_value))
+
+        if child.is_variable_type() or child.is_variable_type() or child.is_parameter_type() or child.is_expression_type():
+            child.set_value(new_value)
+            failed_to_set_value = False
+
+        else:
+            failed_to_set_value = True
+
+        fail_ct += failed_to_set_value
+
+        if failed_to_set_value:
+            warning_msg = 'WARNING: Could not set value of parameter %s' % (key)
+            warnings.warn(warning_msg)
+
+    print('Set %d of %d options to default values (%d failures, see warnings)' %
+        (len(input_dict)-fail_ct, len(input_dict), fail_ct))
```

Comment:
A nit: Python has supported f-strings for a long time now...

Reply:
The muscle memory to do %-formatting is strong, but I'll work on it! Fixed.